sympy/polys/domains/mpelements.py

```diff
@@ -22,10 +22,7 @@ def _set_mpf(self, val):
         prec, rounding = self.context._prec_rounding
         tol = self.context.tol
 
-        if mpf_lt(mpf_abs(val, prec, rounding), tol):
-            self.__mpf__ = fzero
-        else:
-            self.__mpf__ = val
+        self.__mpf__ = val
```

Comment:
@certik, yes, truncation to zero is needed to get degree reduction in polynomial division algorithms. The current approach is most likely wrong (as I recall), but nevertheless, there must be some way to allow polynomial division in `RR`.